File: src/main/Java/boj_step10/boj14215.java
import java.io.*;
import java.util.Arrays;

public class boj14215 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int[] triangle = Arrays.stream(reader.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        triangle = Arrays.stream(triangle).sorted().toArray();
        if (triangle[2] >= triangle[0] + triangle[1]) writer.append(String.valueOf((triangle[0] + triangle[1]) * 2 - 1)).close();
        else writer.append(String.valueOf(Arrays.stream(triangle).sum())).close();
    }
}

File: src/main/boj_step10/boj14215.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var triangle = r.readLine().split(" ").map { it.toInt() }
    triangle = triangle.sorted()
    if(triangle[0] + triangle[1] <= triangle[2]) w.write(((triangle[0] + triangle[1]) * 2 - 1).toString())
    else w.write(triangle.sum().toString())
    w.close()
}